### ISSUE TITLE
Decode failed using akka.http.scaladsl.coding.Deflate

### ISSUE THREAD

**Issue by DSRYhh** [akka/akka-http#1359]
When I try to use akka.http.scaladsl.coding.Deflate to decode HTTP response, there was an exception: `akka.stream.impl.io.ByteStringParser$ParsingException: Parsing failed in step akka.http.scaladsl.coding.DeflateDecompressor$$anon$1$$anon$2@1bb42e8
`. I believe I receive correct bytes, so I try to use `java.util.zip.InflaterInputStream` to decode the HTTP entity with deflate encode, and it works.

This is Akka version (will cause an exception):
```
     
      responseFuture
            .map(Deflate.decodeMessage(_))
            .map(_.entity)
            .flatMap(_.toStrict(1 seconds)(materializer))
            .map(_.data)
            .map(_.utf8String)//this will throw a parse error exception
```

and this is Java version:
```
responseFuture
            .map(_.entity)
            .flatMap(_.toStrict(1 seconds)(materializer))
            .map(_.dataBytes)
            .map(_.runWith(StreamConverters.asInputStream()))
            .map(new InflaterInputStream(_, new Inflater(true)))
            .map(Source.fromInputStream(_).mkString)//this should be fine
```

The entire source code can be seen in [Github gist](https://gist.github.com/DSRYhh/d4a1a8cd828508899b6dcc54693608c7)

**Comment by jrudolph** [comment]
According to https://tools.ietf.org/html/rfc7230#section-4.2.2:

```
   The "deflate" coding is a "zlib" data format [RFC1950] containing a
   "deflate" compressed data stream [RFC1951] that uses a combination of
   the Lempel-Ziv (LZ77) compression algorithm and Huffman coding.

      Note: Some non-conformant implementations send the "deflate"
      compressed data without the zlib wrapper.
```

It seems the server you run against is missing the wrapping, so you need to use `Inflater(nowrap = true)` while akka-http's Deflate implementation uses `Inflater(nowrap = false)` according to the spec.

There might be a way to guess if the zlib wrapper exists or not, if that's the case, we would be open to a PR that adds auto detection.

**Comment by DSRYhh** [comment]
@jrudolph can `Deflate.decodeMessage()` add a optional parameter like `decodeMessage(nowarp : Boolean = false)` to make it work in this case?

**Comment by raboof** [comment]
That sounds neat to me (it'd probably have to be a second method for binary compatibility) - would you be up to creating a PR?

**Comment by jrudolph** [comment]
Fixed by #1522.
